catalog/async/catalog-async-postingestplugin/src/main/java/org/codice/ddf/catalog/async/processingplugin/ProcessingPostIngestPlugin.java

```diff
@@ -0,0 +1,202 @@
+import org.codice.ddf.catalog.async.data.api.internal.ProcessRequest;
+import org.codice.ddf.catalog.async.data.api.internal.ProcessResource;
+import org.codice.ddf.catalog.async.data.api.internal.ProcessUpdateItem;
+import org.codice.ddf.catalog.async.processingframework.api.internal.ProcessingFramework;
+import org.codice.ddf.security.common.Security;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.Response;
+import ddf.catalog.operation.Update;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.ResourceRequestById;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PostIngestPlugin;
+import ddf.catalog.resource.Resource;
+import ddf.catalog.resource.ResourceNotFoundException;
+import ddf.catalog.resource.ResourceNotSupportedException;
+
+/**
+ * The {@code ProcessingPostIngestPlugin} is a {@link PostIngestPlugin} that is responsible for
+ * submitting {@link ProcessRequest}s to the {@link ProcessingFramework}.
+ */
+public class ProcessingPostIngestPlugin implements PostIngestPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProcessingPostIngestPlugin.class);
+
+    private static final String POST_PROCESS_COMPLETE = "post-process-complete";
+
+    private ProcessingFramework processingFramework;
+
+    private CatalogFramework catalogFramework;
+
+    public ProcessingPostIngestPlugin(CatalogFramework catalogFramework,
+            ProcessingFramework processingFramework) {
+        notNull(catalogFramework, "The catalog framework must not be null");
+        notNull(processingFramework, "The processing framework must not be null");
+
+        this.catalogFramework = catalogFramework;
+        this.processingFramework = processingFramework;
+    }
+
+    private static boolean isAlreadyPostProcessed(Response response) {
+        Map<String, Serializable> properties = response.getRequest()
+                .getProperties();
+        if (properties.containsKey(POST_PROCESS_COMPLETE)) {
+            Serializable prop = properties.get(POST_PROCESS_COMPLETE);
+            if (prop instanceof Boolean) {
+                return (boolean) prop;
+            }
+        }
+        return false;
+    }
+
+    private static Map<String, Serializable> putPostProcessCompleteFlagAndGet(
+            Map<String, Serializable> properties) {
+        Map<String, Serializable> newProperties = new HashMap<>(properties);
+        newProperties.put(POST_PROCESS_COMPLETE, true);
+        return newProperties;
+    }
+
+    @Override
+    public CreateResponse process(CreateResponse input) throws PluginExecutionException {
```

Comment:
This may end up being messier because the inputs all have different method names to get the `Metacard`s. For example, `input.getDeletedMetacards()`, `input.getUpdatedMetacards()`, and `input.getCreatedMetacards()`. I cannot provide a Supplier either since the `input.getUpdatedMetacards()` returns a `List<Update>` and the other two return a `List<Metacard>`.